File: C11/백준/Bronze/2231. 분해합/분해합.c
#include <stdio.h>

int main(void){
	
	int N;
	scanf("%d", &N);
	
	for(int i = 1; i <= N; i++){
		int sum = i;
		int temp = i;
		while(temp > 0){
			sum += temp % 10;
			temp /=10;
		}
		if(sum == N){
			printf("%d", i);
			return 0;
		}
	}
	
	printf("%d", 0);
	
	return 0;
}

File: Java/백준/Bronze/2231. 분해합/분해합.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {

        Scanner s = new Scanner(System.in);

        int N = s.nextInt(); //N의 가장 작은 생성자 구하기

        for (int i = 1; i <= N; i++) {
            int sum = i, temp = i;
            while (temp > 0) {
                sum += temp % 10;
                temp /= 10;
            }
            if (sum == N) {
                System.out.print(i);
                return;
            }
        }

        System.out.print(0);

        s.close();
    }
}